framework/test/unit/modules/pos/commands/report_misbehavior.spec.ts

```diff
@@ -600,24 +638,51 @@ describe('ReportMisbehaviorCommand', () => {
 				})
 				.createCommandExecuteContext<PomTransactionParams>(pomCommand.schema);
 
-			when(pomCommand['_tokenMethod'].getAvailableBalance as any)
-				.calledWith(expect.anything(), validator1Address, DEFAULT_LOCAL_ID)
-				.mockResolvedValue(remainingBalance as never);
-
 			await pomCommand.execute(context);
+			const updatedValidator = await validatorSubstore.get(
+				createStoreGetter(stateStore),
+				validator1Address,
+			);
+			const updateStakerData = await stakerSubStore.get(
+				createStoreGetter(stateStore),
+				validator1Address,
+			);
 
+			expect(pomCommand['_tokenMethod'].unlock).toHaveBeenCalledWith(
+				expect.anything(),
+				validator1Address,
+				MODULE_NAME_POS,
+				TOKEN_ID_POS,
+				reportPunishmentReward,
+			);
 			expect(pomCommand['_tokenMethod'].transfer).toHaveBeenCalledWith(
 				expect.anything(),
 				validator1Address,
 				context.transaction.senderAddress,
 				DEFAULT_LOCAL_ID,
 				reportPunishmentReward,
 			);
+			expect(updatedValidator.selfStake).toEqual(
+				misBehavingValidator.selfStake - reportPunishmentReward,
+			);
+			expect(updatedValidator.totalStake).toEqual(
+				misBehavingValidator.totalStake - reportPunishmentReward,
+			);
+			expect(updateStakerData.stakes[0].amount).toEqual(stake.amount - reportPunishmentReward);
+			expect(eligibleValidatorStore['update']).toHaveBeenCalledWith(
+				expect.anything(),
+				validator1Address,
+				oldWeight,
+				updatedValidator,
+			);
 		});
 
-		it('should not reward the sender if validator does not has enough minimum remaining balance', async () => {
-			const remainingBalance = BigInt(100);
-
+		it('should not reward the sender if validator does not have enough self stake', async () => {
```

Comment:
I am not sure I understand what case is checked here. Do you assume that self-stake is 0? If yes, ok, but I am not 100% convinced this is the case. 

In case the self stake is less than 1LSK, but more than 0, then all self stake goes to sender, and this is correctly checked at the next test (line 723).

Reply:
yes, the self-stake is 0 (by default). It should also have been indicated by the test description (which I did now). Thanks for pointing it out!